lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2244,25 +2253,22 @@ class SqlParser(private val ion: IonSystem) : Parser {
         return ParseNode(ARG_LIST, null, items, rem)
     }
 
+
     /**
-     * Checks that the given [Token] list does not have any top-level tokens outside of the top level query. Throws
-     * an error if so.
-     *
-     * Currently only checks for `EXEC`. DDL and DML along with corresponding error codes to be added in the future
-     * (https://github.com/partiql/partiql-lang-kotlin/issues/354).
+     * Validates tree to make sure that the top level tokens are not found below the top level
      */
-    private fun List<Token>.checkUnexpectedTopLevelToken() {
-        for ((index, token) in this.withIndex()) {
-            if (token.keywordText == "exec" && index != 0) {
-                token.err("EXEC call found at unexpected location", PARSE_EXEC_AT_UNEXPECTED_LOCATION)
-            }
+    private fun validateTopLevelNodes(node: ParseNode, level: Int) {
+        // Note that for DML operations, top level parse node is of type 'FROM'. Hence the check level > 1
+        if (node.type.isTopLevelType && ((node.type.isDml && level > 1) || (!node.type.isDml && level > 0))) {
```

Comment:
nit: This can be a little hard to understand with so many logic symbols. Can this be simplified or broken into multiple if-statements?

Reply:
It will make the code redundant. I also thought the conditional statement is complicated initially  but on the second glance it is not that complicated. 

```
if (node.type.isTopLevelType && level > 0) {
    if (node.type.isDml) {
        if (level > 1) {
            // throw error
        }
    } else {
        // throw error
    }
}
```

Reply:
Changed as mentioned above.